cellrank/tl/estimators/mixins/decomposition/_eigen.py

```diff
@@ -0,0 +1,378 @@
+        ncv
+            Number of Lanczos vectors generated.
+
+        Returns
+        -------
+        Nothing, just updates the following field:
+
+            - :attr:`eigendecomposition` - %(eigen.full_desc)s
+        """
+
+        def get_top_k_evals() -> np.ndarray:
+            return D[np.flip(np.argsort(D.real))][:k]
+
+        start = logg.info("Computing eigendecomposition of the transition matrix")
+
+        if issparse(self.transition_matrix):
+            logg.debug(f"Computing top `{k}` eigenvalues of a sparse matrix")
+            D, V_l = eigs(self.transition_matrix.T, k=k, which=which, ncv=ncv)
+            if only_evals:
+                self._write_eigendecomposition(
+                    {
+                        "D": get_top_k_evals(),
+                        "eigengap": _eigengap(get_top_k_evals().real, alpha),
+                        "params": {"which": which, "k": k, "alpha": alpha},
+                    },
+                    time=start,
+                )
+                return
+            _, V_r = eigs(self.transition_matrix, k=k, which=which, ncv=ncv)
+        else:
+            logg.warning(
+                "This transition matrix is not sparse, computing full eigendecomposition"
+            )
+            D, V_l = np.linalg.eig(self.transition_matrix.T)
+            if only_evals:
+                self._write_eigendecomposition(
+                    {
+                        "D": get_top_k_evals(),
+                        "eigengap": _eigengap(D.real, alpha),
+                        "params": {"which": which, "k": k, "alpha": alpha},
+                    },
+                    time=start,
+                )
+                return
+            _, V_r = np.linalg.eig(self.transition_matrix)
+
+        # Sort the eigenvalues and eigenvectors and take the real part
+        logg.debug("Sorting eigenvalues by their real part")
+        p = np.flip(np.argsort(D.real))
+        D, V_l, V_r = D[p], V_l[:, p], V_r[:, p]
+        e_gap = _eigengap(D.real, alpha)
+
+        pi = np.abs(V_l[:, 0].real)
+        pi /= np.sum(pi)
+
+        self._write_eigendecomposition(
+            {
+                "D": D,
+                "stationary_dist": pi,
+                "V_l": V_l,
+                "V_r": V_r,
+                "eigengap": e_gap,
+                "params": {"which": which, "k": k, "alpha": alpha},
+            },
+            time=start,
+        )
+
+    @d.dedent
+    def plot_spectrum(
+        self,
+        n: Optional[int] = None,
+        real_only: bool = False,
+        show_eigengap: bool = True,
+        show_all_xticks: bool = True,
+        legend_loc: Optional[str] = None,
+        title: Optional[str] = None,
+        marker: str = ".",
+        figsize: Optional[Tuple[float, float]] = (5, 5),
+        dpi: int = 100,
+        save: Optional[Union[str, Path]] = None,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Plot the top eigenvalues in real or complex plane.
```

Comment:
```suggestion
        """Plot the top eigenvalues in real or complex plane.
```

Reply:
@WeilerP I don't think I will be changing the docstrings to this format, I prefer what we have now.